python/ray/dataframe/dataframe.py

```diff
@@ -3858,9 +3858,48 @@ def unstack(self, level=-1, fill_value=None):
 
     def update(self, other, join='left', overwrite=True, filter_func=None,
                raise_conflict=False):
-        raise NotImplementedError(
-            "To contribute to Pandas on Ray, please visit "
-            "github.com/ray-project/ray.")
+        """Modify DataFrame in place using non-NA values from passed
+        DataFrame. Aligns on indices
+
+        Args:
+            other: DataFrame, or object coercible into a DataFrame
+            join: {'left'}, default 'left'
+            overwrite: If True then overwrite values for common keys in frame
+            filter_func: Can choose to replace values other than NA.
+            raise_conflict: If True, will raise an error if the DataFrame and
+                other both contain data in the same place.
+
+        Returns:
+            None
+        """
+        if raise_conflict:
+            raise NotImplementedError(
+                "To contribute to Pandas on Ray, please visit "
+                "github.com/ray-project/ray.")
+
+        if join != 'left':  # pragma: no cover
+            raise NotImplementedError("Only left join is supported")
+
+        if not isinstance(other, DataFrame):
+            other = DataFrame(other)
+
+        def _update_part(x, y, axis, level):
+            new_df = x.copy()
+            new_df.update(
```

Comment:
Inner partition index does not match outer index in our architecture. This will not work here, you will need to handle the difference in the outer and inner index when calling the underlying pandas `update` function.

Reply:
Ah, I see. I noticed in the pandas source, that they align the indexes up before performing the update. Do you think it would be best for me to wait unitl `reindex`/`reindex_like` get's merged in?

Reply:
Actually, i'll look at devin's join to see if I can do it without using `reindex`

Reply:
There is a `_reindex_helper` in utils. It is only used for join currently.